pkg/convert/pot.go

```diff
@@ -110,47 +74,209 @@ func FromPot(b []byte) (model.Messages, error) {
+		}
+
+		if len(pluralLines) == 1 {
+			if _, err = fmt.Fprintf(b, "msgstr[%d] \"%s\"\n", i, pluralLines[0]); err != nil {
+				return fmt.Errorf("write %s: %w", tag, err)
+			}
+		} else {
+			if _, err = fmt.Fprintf(b, "msgstr[%d] \"\"\n", i); err != nil {
+				return fmt.Errorf("write %s: %w", tag, err)
+			}
+			if err = writeMultiline(b, tag, pluralLines); err != nil {
+				return fmt.Errorf("write plural multiline: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func writeMultiline(b *bytes.Buffer, tag poTag, lines []string) error {
 	for _, line := range lines {
 		if strings.HasSuffix(line, "\\n") {
-			_, err = fmt.Fprint(b, "\""+line+"\""+"\n")
-			if err != nil {
+			if _, err := fmt.Fprint(b, "\""+line+"\""+"\n"); err != nil {
 				return fmt.Errorf("write %s: %w", tag, err)
 			}
 		} else {
-			_, err = fmt.Fprint(b, "\""+line+"\\n\""+"\n")
-			if err != nil {
+			if _, err := fmt.Fprint(b, "\""+line+"\\n\""+"\n"); err != nil {
 				return fmt.Errorf("write %s: %w", tag, err)
 			}
 		}
 	}
 
 	return nil
 }
+
+func getPoTagLines(str string, tag poTag) (error, []string) {
+	encodedStr, err := json.Marshal(str) // use JSON encoding to escape special characters
+	if err != nil {
+		return fmt.Errorf("marshal %s: %w", tag, err), nil
+	}
+
+	encodedStr = encodedStr[1 : len(encodedStr)-1] // trim quotes
+	lines := strings.Split(string(encodedStr), "\\n")
+
+	// Remove the empty string element. The line is empty when splitting by "\\n" and "\n" is the last character in str.
+	if lines[len(lines)-1] == "" {
+		lines = lines[:len(lines)-1]
+		lines[len(lines)-1] = lines[len(lines)-1] + "\\n" // add the "\n" back to the last line
+	}
+
+	return nil, lines
+}
+
+func extractPluralForms(plural string) ([]string, error) {
+	if plural == "" {
+		return nil, fmt.Errorf("plural is empty")
+	}
+
+	plural = strings.ReplaceAll(plural, "{$count}", "%d")
+	// Use regular expressions to extract the plural forms from the plural finds string in {}
+	re := regexp.MustCompile(`when [\d*]+ \{([^}]+)\}`)
+	matches := re.FindAllStringSubmatch(plural, -1)
+
+	if len(matches) == 0 {
+		return nil, fmt.Errorf("no plural forms found in plural")
+	}
+
+	// Create a slice of strings to hold the extracted plural forms
+	pluralForms := make([]string, 0, len(matches))
+
+	// Loop through the matches and extract the plural form from each one
+	for _, match := range matches {
+		pluralForms = append(pluralForms, match[1])
+	}
+
+	return pluralForms, nil
+}
+
+func convertToFormatMessage(plurals []string) string {
+	var sb strings.Builder
+
+	sb.WriteString("match {$count :number}\n")
+
+	for i, plural := range plurals {
+		line := strings.ReplaceAll(strings.TrimSpace(plural), "%d", "{$count}")
+
+		switch {
+		case i == 0:
+			sb.WriteString(fmt.Sprintf("when 1 {%s}\n", line))
+		case i == len(plurals)-1:
+			sb.WriteString(fmt.Sprintf("when * {%s}\n", line))
+		default:
+			sb.WriteString(fmt.Sprintf("when %d {%s}\n", i+1, line))
+		}
+	}
```

Comment:
I think the code with 2 cases and 1 default is more readable because it presents conditions more explicitly.

Reply:
Maybe you are right, but I just don't like repetition because here, `case i==0` and `default` does the same thing

Nevertheless `switch i {}` needs to be used here